scripts/match-2-yar.py

```diff
@@ -0,0 +1,868 @@
+import capa.render.utils as rutils
+import capa.render.verbose
+import capa.features.freeze
+import capa.render.result_document as rd
+from capa.features.common import OS_AUTO
+from capa.features.extractors.dnfile.extractor import DnfileFeatureExtractor
+
+try:
+    import yaramod
+    from mkyara import YaraGenerator
+    from capstone import CS_MODE_32, CS_MODE_64, CS_ARCH_X86, CS_OPT_SYNTAX_INTEL, Cs
+except ImportError:
+    print(
+        """\nFailed to import a module try installing required Python libraries with the following:
+pip install mkyara yaramod
+"""
+    )
+    sys.exit(1)
+
+
+logger = logging.getLogger("capa.match-2-yar")
+
+
+# Vivisect Related Classes and Functions
+
+
+class BufferCanvas(MemoryCanvas):
+    """Subclass of Vivisect Memory canvas that captures
+    disassemlby output as a string rather than printing to std.out
+    """
+
+    output = ""
+
+    def addText(self, text, tag=None):
+        """Overwriting the method responsible for writing to std.out"""
+        self.output += text
+
+
+def get_disassembly_output(vw, va, size):
+    """Get Vivisect's disassembly view for a given virtual addresss and size
+
+    Args:
+        vw: Vivisect Workspace
+        va: Virtual Address to start disassembling from
+        size: size in bytes to disassemble
+
+    Returns:
+        str: String containing vivisect's disassembly output
+    """
+    rend = WorkspaceRenderer(vw)
+    mcav = BufferCanvas(vw)
+    mcav.renderMemory(va, size, rend=rend)
+    return mcav.output
+
+
+def get_comment_for_func(vw, funcva):
+    """Get a CodeFeature comment for a function
+
+    This function gets the size of a function and
+    uses that to get a dump of the function disassembly
+    with get_dissasembly_output
+
+    Args:
+        vw: Vivisect Workspace
+        funcva: Virtual Address of function to analyze
+
+    Returns:
+        str: String containing disassembly output for a function
+    """
+    funcsize = get_function_size(vw, funcva)
+    return get_disassembly_output(vw, funcva, funcsize)
+
+
+def get_comment_for_cb(vw, va):
+    """Get a CodeFeature comment for a Code Block
+
+    This function gets the size of a code block and
+    uses that to get a dump of the code block disassembly
+    with get_dissasembly_output
+
+    Args:
+        vw: Vivisect Workspace
+        va: Virtual Address of Codeblock to analyze
+
+    Returns:
+        str: String containing disassembly output for a function
+    """
+    cb = vw.getCodeBlock(va)
+    cbva, cbsize, cbfunc = cb
+    return get_disassembly_output(vw, cbva, cbsize)
+
+
+def get_function_size(vw, funcva):
+    """Return the size of a function based on vivisect analysis
+
+    Args:
+        vw: Vivisect Workspace
+        funcva: Virtual Address of function to analyze
+
+    Returns:
+        int: size of the function
+    """
+    fsize = 0
+    if funcva not in vw.getFunctions():
+        funcva = vw.getFunction(funcva)
+        if funcva is None:
```

Comment:
is this the same check done twice?

Reply:
The logic was a little funky here. getFunction(va) will get the function that a given virtual address is a part of.

So originally it was saying is the provided funcva a function, if not what function does it belong to. But that getFunction might overwrite the value of funcva in a weird way.

I updated to just use getFunction and add an explicit check to see if the returned effective_funcva is the same as the provided funcva

Reply:
Ah, gotcha. Thanks for the clarification and comments.